packages/story-editor/src/app/history/reducer.ts

```diff
@@ -13,6 +13,16 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+/**
+ * Internal dependencies
+ */
+import type {
+  ReducerState,
+  ReducerProps,
+  HistoryEntry,
+} from '../../types/historyProvider';
+
 export const SET_CURRENT_STATE = 'set_state';
 export const CLEAR_HISTORY = 'clear';
 export const REPLAY = 'replay';
```

Comment:
Do you mean creating an enum type for these instead of using  `typeof X`? As it currently is, we won't have to keep the enum AND the constants both updated.
Also, when using enum with TypeScript, it won't work with the reducer the same way -- currently, TS understand that if it's `set_state` then specific params are expected, if it's `clear` then something else, etc. In case of `enum`, it doesn't seem to understand that. I'll double check again but at least the last time I tried, it didn't work like that.

Reply:
All ok now after specifying in Slack with Morten,